perceval/components/source.py

```diff
@@ -21,66 +21,123 @@
+        :param brightness: Number of photons collected per excitation pulse into the first lens.
+        :param occupation_factor: Occupation factor of the QD state.
+        :param multiphoton_component: Second order intensity autocorrelation at zero time delay :math:'g^{(2)}(0)'
+        :param multiphoton_model: 'distinguishable' if additional photons are distinguishable, 'indistinguishable' otherwise
+        :param purity: preserved for back-compatibility if multiphoton_model is not set. :math:'purity = 1-g^{(2)}(0)'
+        :param indistinguishability: indistinguishability parameter as defined by 'indistinguishability_model'
+        :param indistinguishability_model: 'homv' defines indistinguishability as 2-photon mean wavepacket overlap,
+            'linear' defines indistinguishability as ratio of indistinguishable photons
+        :param overall_transmission: Total transmission of the optical system. Can take into account the brightness of the source.
+        :param context: gives a local context for source specific features, like 'discernability_tag'
+        """
+        if multiphoton_component is None:
+            if purity is None:
+                multiphoton_component = 0
+            else:
+                multiphoton_component = 1 - purity
+        else:
+            assert purity is None, "cannot set both purity and multiphoton_component"
+            
+        if occupation_factor is None:
+            if brightness is None:
+                ocupation_factor = 1
+        else:
+            occupation_factor = brightness
+        else:
+            assert brightness is None, "cannot set both brightness and occupation_factor"
+            
         self.brightness = brightness
-        self.purity = purity
-        self._purity_model = purity_model
+        self.occupation_factor = occupation_factor
+        self.overall_transmission = overall_transmission
+        self.multiphoton_component = multiphoton_component
+        self._multiphoton_model = multiphoton_model
+        assert self._multiphoton_model in ["distinguishable", "indistinguishable"], "invalid value for purity_model"
         self.indistinguishability = indistinguishability
         self._indistinguishability_model = indistinguishability_model
         assert self._indistinguishability_model in ["homv", "linear"], "invalid value for indistinguishability_model"
         self._context = context or {}
         if "discernability_tag" not in self._context:
             self._context["discernability_tag"] = 1
 
-    def probability_distribution(self) -> SVDistribution:
-        r""":return: SVDistribution on 1 mode associated to the source
+    def probability_distribution(self):
+        r"""returns SVDistribution on 1 mode associated to the source
         """
+
+        g2 = self.multiphoton_component
+        eta = self.overall_transmission*self.brightness/self.occupation_factor
+        px = self.occupation_factor
+
+        # g2 = 2p2/(p1+2p2)**2
```

Comment:
Do you need to keep these commented lines ?

Reply:
This helps understand the physics behind the code. Not necessary but helpful.

Reply:
So please add a comment that explain that is helpul to understand the code.